testframework/src/main/java/net/neoforged/testframework/summary/SummaryFormatter.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.testframework.summary;
+
+import org.slf4j.Logger;
+
+@FunctionalInterface
+public interface SummaryFormatter {
```

Comment:
I would expect a “formatter” to be a unary operator, if it's also meant to handle writing then `format` is a misleading name. `Dumper` is much better.